source/doc-pages/guides/basics/repositories.md

```diff
@@ -0,0 +1,309 @@
+repositories rather than accessing lower-level query DSLs.
+
+``` ruby
+class Users < ROM::Relation[:sql]
+  def listing
+    select(:id, :name, :email, :created_at).order(:name, :id)
+  end
+
+  def registered_after(timestamp)
+    where { created_at > timestamp }
+  end
+end
+
+class UserRepository < ROM::Repository::Base
+  relations :users
+
+  def new_users(timestamp)
+    users.listing.registered_after(timestamp)
+  end
+end
+
+user_repo.new_users(Time.new(2015))
+```
+
+This way you encapsulate specific queries and domain-specific relation views. Your
+application sees simple structs and has no idea how they are fetched and instantiated.
+
+## Relation Views
+
+Every method that you define in a relation represents a specific relation **view**.
+For relation views that you re-use to compose more complex relations you can
+explicitly define their structure with `view` plugin:
+
+``` ruby
+class Users < ROM::Relation[:sql]
+  view(:listing, [:id, :name, :email, :created_at]) do
+    select(:id, :name, :email, :created_at).order(:name, :id)
+  end
+
+  def registered_after(timestamp)
+    where { created_at > timestamp }
+  end
+end
+```
+
+Thanks to this feature we can be explicit about the data structures our relations
+return which comes with the benefit of auto-mapping and better introspection
+capabilities that can be used to build even more advanced features.
+
+<div class="well">
+The view feature will be enhanced with type annotations so that it will be possible
+to automatically map database-specific types into domain-specific types.
+</div>
+
+### Base View Plugin
+
+ROM repository provides an SQL relation plugin called `base_view` which defines
+a `base` relation view for you which includes all the column names and order by
+the primary key:
+
+``` ruby
+class Users < ROM::Relation[:sql]
+  def filter(conditions)
+    base.where(conditions)
+  end
+end
+```
+
+## Composing Relations
+
+Every repository can access multiple relations. Thanks to this you can compose
+them however you want and get even very complex aggregates back. There's no need
+to define `associations` with complicated configuration logic. You simply define
+methods on your repository objects:
+
+``` ruby
+class UserRepository < ROM::Repository::Base
+  relations :users, :tasks
+
+  def with_tasks(id)
+    users.by_id(id).combine_children(many: tasks)
+  end
+end
+
+user_repo.with_tasks.to_a
+# [#<ROM::Struct[User] id=1 name="Jane" tasks=[#<ROM::Struct[Task] id=2 user_id=1 title="Jane Task">]>, #<ROM::Struct[User] id=2 name="Joe" tasks=[#<ROM::Struct[Task] id=1 user_id=2 title="Joe Task">]>]
+```
+
+What you see here is an example usage of `auto_combine` plugin.
+
+### Composition Plugins
+
+ROM Repository is built on top of lower-level relation interface of ROM. It ships
+with a couple of plugins that are simple syntax sugar using Relation `combine`
+interface under the hood.
+
+The beauty of this approach is that even most complex scenarios are handled by
```

Comment:
The beauty of this approach is that even the most complex scenarios are handled by
the very same composition interface. There's nothing special going on here, we're
simply using relations and their views. When default behavior is not satisfactory,
you can define your own relation view, improve the query yourself and it will work
exactly the same in terms of the internal machanics.